ports/esp32/.gitignore

```diff
@@ -0,0 +1 @@
+build-*/
```

Comment:
How about we just add this to the top-level .gitignore?  That would simplify other ports, eg could remove `ports/stm32/.gitignore` and simplify `ports/unix/.gitignore`.

Reply:
Done.